File: src/main/java/io/tripled/social/client/presentation/controller/ReadController.java
package io.tripled.social.client.presentation.controller;

import io.tripled.social.client.application.ReadMessagesUseCase;
import io.tripled.social.client.presentation.Output;

import java.util.regex.Matcher;
import java.util.regex.Pattern;

public class ReadController extends AbstractController {

  private ReadMessagesUseCase readMessagesApplicationService;

  public ReadController(ReadMessagesUseCase service) {
    this.readMessagesApplicationService = service;
  }

  @Override
  protected Pattern getPattern() {
    return Pattern.compile("(\\w+)");
  }

  @Override
  public void execute(String inputLine, Output output) {
    Matcher matcher = getPattern()
        .matcher(inputLine.trim());

    if (matcher.matches()) {
      String userName = matcher.group(1).trim();

      output.print(readMessagesApplicationService.readMessage(userName));
    }
  }
}

File: src/main/kotlin/io/tripled/social/client/presentation/controller/ReadController.kt
package io.tripled.social.client.presentation.controller

import io.tripled.social.client.application.ReadMessagesUseCase
import io.tripled.social.client.presentation.Controller
import io.tripled.social.client.presentation.Output
import io.tripled.social.client.presentation.request.ReadCliRequest

class ReadController(private val readMessagesApplicationService: ReadMessagesUseCase) : Controller<ReadCliRequest> {
    override fun execute(request: ReadCliRequest, output: Output) {
        output.print(readMessagesApplicationService.readMessage(request.userName))
    }
}
